### PR TITLE
feat: Update python version to 3.12, Update PREPARE_ENVIRONMENT.md, F…

### PR DESCRIPTION
**Pull request by polkovnikvasin**
…AQ.md docs and add new doc HOW_TO_ADD_NEW_PROVIDERS.md

BREAKING CHANGE: all dependencies has been updated due to new python version. docker container and dependencies should be updated to use this change.
#10

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -4,36 +4,36 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
```
Comment (vsuzdaltsev) [inline Pipfile:6]: alphabetical order pls

File: requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+black==24.10.0
+dictdiffer==0.9.0
+ipdb==0.13.13
+pylint==3.3.1
+pytest==8.3.3
+pytest-xdist==3.6.1
```
Comment (vsuzdaltsev) [inline requirements-dev.txt:6]: new line pls

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+GitPython==3.1.43
+hvac==2.3.0
+kubernetes==31.0.0
+pygithub==2.5.0
+pyOpenSSL==24.2.1
+python-on-whales==0.73.0
+pyyaml==6.0.2
+slack_sdk==3.33.3
+tomli==2.1.0
+zipp==3.21.0
```
Comment (vsuzdaltsev) [inline requirements.txt:23]: newline pls

**Review by vsuzdaltsev — CHANGES_REQUESTED** [review]

File: Pipfile (hunk 1 above)
Comment (polkovnikvasin) [reply Pipfile:6]: done

**Review by vsuzdaltsev — APPROVED** [review]

**Pull request merged**